tests/CMakeLists.txt

```diff
@@ -0,0 +1 @@
+add_subdirectory(kokkos-utils)
```

Comment:
I'm convinced that the include files should be in a "repeated" `kokkos-utils` subdirectory.

I would not do that for any other file though. So to me, tests should reside in `tests/`, not `tests/kokkos-utils`.